src/felix86/hle/signals.cpp

```diff
@@ -183,8 +183,20 @@ u64 get_actual_rip(BlockMetadata& metadata, u64 host_pc) {
 #define REG_PC 0 // risc-v stores it in gpr 0
 #endif
 
+x64_rt_sigframe* setupFrame32(RegisteredSignal& signal, int sig, ThreadState* state, const u64* host_gprs, const u64* host_fprs,
```

Comment:
Since we only use the RIP from the frame, to avoid different return types let's make these return u64 and do `u64 old_rip = setupFrame` instead of `u64 old_rip = frame->uc.uc_mcontext.gregs[REG_RIP];`